Modules/signalmodule.c

```diff
@@ -530,9 +530,14 @@ signal_strsignal_impl(PyObject *module, int signalnum)
         return NULL;
     }
 
-#ifdef MS_WINDOWS
-    /* Custom redefinition of POSIX signals allowed on Windows */
+#ifndef HAVE_STRSIGNAL
     switch (signalnum) {
+        /* Though being a UNIX, HP-UX does not provide strsignal(3) */
+#ifndef MS_WINDOWS
+        case SIGHUP:
```

Comment:
This list was just a redefinition of POSIX signals supported by Windows. I think we need to check whether there is more than one signal to add to this list for other Unix systems.

Reply:
Well, I can go through `signal.h` on HP-UX and provide those, but this is out of scope of this PR because the motivation for this one is just to make this module properly compile on HP-UX. Is that OK for you?

Reply:
@berkerpeksag I went through `signal.h`. There is a bus load of signals, some wrapped with ifdefs. I don't think that HP-UX is worth the effort for now just to have a nice string. If so, this has to be a separate PR.

Reply:
We definitely can't maintain a full list of signals for every platform we support, so I agree with you. I was wondering whether there are other signals to include to make `strsignal()` useful for most common cases.

I just checked http://nixdoc.net/man-pages/hp-ux/man5/signal.5.html and I think the most common unlisted signals are SIGALRM, SIGPIPE, SIGQUIT, and SIGCHLD.

Since SIGINT and SIGTERM is already in the list, I'd expect SIGQUIT to be there as well.

Reply:
So, I added all those. checked FreeBSD and glibc implementation + manpages and took the strings (`_sys_siglist[]`) out of them. They are identical. I am sure whether the tests should be extented because the Open Group says about `strsignal(3)` that the return value is implementation-defined.

@berkerpeksag WDYT?

Reply:
Awesome, thanks! No need to update the tests.

I will take another look at this and merge tonight.